### PR TITLE
Modernize check log

### PR DESCRIPTION
**Pull request by RincewindsHat**
Some cleanup stuff and shellcheck corrections.
Also:
 * adds `-a|--all` option to print all matching lines,
 * add  `-p|--perl-regex` option to use grep with perl style regex
 * add  `-e|--extended-regex` option to use grep with extended regex style

### CODE REVIEW

File: plugins-scripts/check_log.sh [hunk 1]
```diff
@@ -60,20 +59,25 @@
...
+	echo ""
+	echo "Other parameters:"
+	echo "	-a|--all : Print all matching lines"
+	echo "	-p|--perl-regex : Use perl style regular expressions in the query"
+	echo "	-e|--extended-regex : Use extended style regular expressions in the query (not necessary for GNU grep)"
 }
 
 print_help() {
-    print_revision $PROGNAME $REVISION
+    print_revision "$PROGNAME" $REVISION
```
Comment (sni) [inline plugins-scripts/check_log.sh:80]: i guess you want to put quotes over all variables?

File: plugins-scripts/check_log.sh [hunk 2]
```diff
@@ -100,111 +104,159 @@ while test -n "$1"; do
...
             ;;
         -h)
             print_help
-            exit $STATE_OK
+            exit "$STATE_OK"
             ;;
         --version)
-            print_revision $PROGNAME $REVISION
-            exit $STATE_OK
+            print_revision "$PROGNAME" $REVISION
```
Comment (sni) [inline plugins-scripts/check_log.sh:114]: same here

File: plugins-scripts/check_log.sh [hunk 3]
```diff
@@ -100,111 +104,159 @@ while test -n "$1"; do
...
+            exit "$STATE_OK"
             ;;
         -V)
-            print_revision $PROGNAME $REVISION
-            exit $STATE_OK
+            print_revision "$PROGNAME" $REVISION
```
Comment (sni) [inline plugins-scripts/check_log.sh:118]: and here

**Review by sni — CHANGES_REQUESTED** [review]
looks good, just a minor change request :-)

File: plugins-scripts/check_log.sh (hunk 1 above)
Comment (RincewindsHat) [reply plugins-scripts/check_log.sh:80]: I do and localy they are there oO. Also they are in my branch...

**Comment by RincewindsHat** [comment]
Updated it

**Comment by RincewindsHat** [comment]
@sni

**Review by sni — APPROVED** [review]

**Pull request merged**